submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestIndexer.kt

```diff
@@ -0,0 +1,44 @@
+package ac.uk.ebi.biostd.submission.submitter.request
+
+import ac.uk.ebi.biostd.persistence.common.model.RequestStatus.INDEXED
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequestFile
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestFilesPersistenceService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.ExtSubmission
+import mu.KotlinLogging
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
+import uk.ac.ebi.extended.serialization.service.fileSequence
+import java.util.concurrent.atomic.AtomicInteger
+
+private val logger = KotlinLogging.logger {}
+
+class SubmissionRequestIndexer(
+    private val extSerializationService: ExtSerializationService,
+    private val requestService: SubmissionRequestPersistenceService,
+    private val filesRequestService: SubmissionRequestFilesPersistenceService,
+) {
+    fun indexRequest(accNo: String, version: Int) {
+        logger.info { "Started loading pending request accNo='$accNo', version='$version'" }
+        val request = requestService.getPendingRequest(accNo, version)
+        logger.info { "Finished loading pending request accNo='$accNo', version='$version'" }
+
+        indexSubmissionFiles(request.submission)
+        val totalFiles = filesRequestService.getSubmissionRequestFiles(accNo, version, 0).count().toInt()
```

Comment:
this perform really bad as we are loading every single item only to count it.  We can actually do something like 

```
        extSerializationService.fileSequence(sub)
            .onEach { indexFile(it, index.incrementAndGet()) }
            .count()       
```